### PR TITLE
🔒️ Adds codeql-analysis cron job

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- Activates cron job that analyzes repo code for vulnerabilities (every monday at 5:26am)
- Fixes two clear-text logging of sensitive info 
   -  NOTE: as a good practice always use ``SecretStr`` in pydantic models for objects that are logged

More details:
- [codeQL]( https://codeql.github.com/docs/codeql-overview/)
- [GH-Action](https://github.com/github/codeql-action)
- [LGTM](https://lgtm.com/#explore)
- [codeql-config](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-a-custom-configuration-file)

To see some of the job runs, check the ["Actions" tab and select CodeQL workflow](https://github.com/ITISFoundation/osparc-simcore/actions/workflows/codeql-analysis.yml) 

Screen Shot 2022-03-15 at 22 56 00

## Related issue/s

- ITISFoundation/osparc-issues#509

## Checklist

- [x] scan only src/ folders
- [x] leave only cron-job and disable push+pull-request
- [x] analyze code and apply fixes
- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Database migration script? ``cd packages/postgres-database``, ``make setup-commit``, ``sc-pg review -m "my changes"``
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# codeQL       : https://codeql.github.com/docs/codeql-overview/
+# GH-Action    : https://github.com/github/codeql-action
+# LGTM         : https://lgtm.com/#explore
+# codeql-config: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-a-custom-configuration-file
+#
+name: "CodeQL"
+
+on:
+  schedule:
+    - cron: "26 5 * * 1"
```
Comment (mguidon) [inline .github/workflows/codeql-analysis.yml:11]: Any particular reason for the time?

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def get_dynamic_sidecar_spec(
                 )
             )
 
-    endpint_spec = {}
+    endpoint_spec = {}
```
Comment (mguidon) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py:132]: i would never trade a pint for a point

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (pcrespov) [reply .github/workflows/codeql-analysis.yml:11]: I wanted to be a time where we normally are not running CI and did not want to be on o'clock , half-past or quarter ...

File: .github/codeql/codeql-config.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: "ospac-simcore CodeQL config"
+
+disable-default-queries: false
```
Comment (mguidon) [inline .github/codeql/codeql-config.yml:3]: So what are we scanning for by default?

**Review by mguidon — APPROVED** [review]
Very nice tool!

File: .github/codeql/codeql-config.yml (hunk 3 above)
Comment (pcrespov) [reply .github/codeql/codeql-config.yml:3]: Check the links above RT... :-)  ... but basically thinks like plotting sensitive info (e.g. he was able to trace in the code and detect that a password was printed in the logs ... was quite impressive BTW)

File: .github/codeql/codeql-config.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  - services/catalog/src/simcore_service_catalog
+  - services/dask-sidecar/src/simcore_service_dask_sidecar
+  - services/datcore-adapter/src/simcore_service_datcore_adapter
+  - services/director-v2/src/simcore_service_director_v2
+  - services/dynamic-sidecar/src/simcore_service_dynamic_sidecar
+  - services/storage/src/simcore_service_storage
+  - services/web/server/src/simcore_service_webserver
+  - "**/src"
+  - "**/source"
+paths-ignore:
```
Comment (GitHK) [inline .github/codeql/codeql-config.yml:25]: so I guess we never care about tests

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py (hunk 2 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py:132]: I might have been drunk here!

**Review by GitHK — APPROVED** [review]

File: .github/codeql/codeql-config.yml (hunk 4 above)
Comment (pcrespov) [reply .github/codeql/codeql-config.yml:25]: we do care ( al ot) about tests :-). ... but yes, I do not see how security rules would be useful in those

**Pull request merged**